src/psyclone/dynamo0p3.py

```diff
@@ -2896,29 +2900,30 @@ def __init__(self, node):
         # Create lists of real and integer scalar arguments
         self._real_scalar_names = {}
         self._int_scalar_names = {}
+        for intent in FORTRAN_INTENT_NAMES:
+            self._real_scalar_names[intent] = []
+            self._int_scalar_names[intent] = []
         if self._invoke:
-            real_scalars = self._invoke.unique_declns_by_intent("gh_real")
-            int_scalars = self._invoke.unique_declns_by_intent("gh_integer")
-            for intent in FORTRAN_INTENT_NAMES:
-                self._real_scalar_names[intent] = [arg.declaration_name for arg
-                                                   in real_scalars[intent]]
-                self._int_scalar_names[intent] = [arg.declaration_name for arg
-                                                  in int_scalars[intent]]
+            for scalar_type in LFRicArgDescriptor.VALID_SCALAR_NAMES:
+                scalar_dict = self._invoke.unique_declns_by_intent(scalar_type)
+                for intent in FORTRAN_INTENT_NAMES:
+                    for arg in scalar_dict[intent]:
+                        declname = arg.declaration_name
+                        if arg.intrinsic_type == "real":
+                            self._real_scalar_names[intent].append(declname)
+                        if arg.intrinsic_type == "integer":
+                            self._int_scalar_names[intent].append(declname)
```

Comment:
As this stands, we need a final `elif` to catch cases where the `intrinsic_type` is something we don't support.

Reply:
Not any more, as noted in one of the above comments. The invalid intrinsic type is now caught when setting up `DynKernelArgument.intrinsic_type` property that creates intrinsic type from `dynamo0p3.MAPPING_DATA_TYPES` and checks there if the mapping is invalid.